src/Db.php

```diff
@@ -17,8 +17,10 @@ public static function getInstance(): Db {
   private function __construct() {
     $this->config = parse_ini_file($this->settings_file);
     $options = array(
-      'idle_timeout_micros' => 200000,
+      'idle_timeout_micros' => 2000000,
```

Comment:
> 'idle_timeout_micros' => 2000000,

**Can you explain the justification for the value chosen?**

The default value is `4000000` microseconds or 4 seconds.  The value selected here is `2000000` or 2 seconds, which is lower than the default, though higher than what is currently set within the project.

Given we are using an `expiration_policy` of `IdleTime`, this value will determine when the connections in the pool are destroyed.  While this does not directly have any relation to persistent connections, the implications, in this case, are similar.  Given the amount of caching, database queries should be rarer than they previously were within the platform, and a value of 0.2 seconds (the old value) is likely too low, resulting in an increased volume of connection overhead.  However, 2 seconds might also be too little as well.  It seems likely that the new Cache/DB design would benefit from an increased value here.

Do you have any test data that provides merit to this particular value?

Reply:
I think with 4 seconds would be ok, but 0.2 was too low and saw that the requests were generating way too many new connections. Also, a high value (like 10s) will keep the connections for a longer time which could also bring some issues if the database doesn't have a lot of available connections in the pool.

I think we should set this to a number we feel confortable but not leave the default of the framework. If this changes tomorrow to a lower value we'll see a lot of performance issues imo.